x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -489,6 +503,28 @@ protected void onAbort() {
             logger.info("Data frame transform [" + transform.getId() + "] received abort request, stopping indexer");
             shutdown();
         }
+
+        @Override
+        protected void createCheckpoint() {
+            CountDownLatch latch = new CountDownLatch(1);
+
+            transformsCheckpointService.getCheckpoint(transformConfig, currentCheckpoint.get() + 1,
+                    new LatchedActionListener<>(ActionListener.wrap(checkpoint -> {
+                        transformsConfigManager.putTransformCheckpoint(checkpoint, ActionListener.wrap(putCheckPointResponse -> {
```

Comment:
Does this actually work? I am not sure how `LatchedActionListener` works with nested `ActionListener`s like this. 

As an aside, since will be called out of the generic thread pool via the executor, it could just use `PlainFuture` or something and synchronously call `get(timeout)`. Unsure if that will simplify it any.